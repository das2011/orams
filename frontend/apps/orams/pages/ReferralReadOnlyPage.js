import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter, Switch, Route } from 'react-router-dom'
import ReferralDetails from 'orams/components/ReferralDetails/ReferralDetails'
import { loadReferralData } from 'orams/actions/referralActions'

export class ReferralReadOnlyPage extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  componentDidMount() {
    this.props.loadReferralData(this.props.match.params.id)
  }

  render() {
    const { match } = this.props

    return (
      <Switch>
        <Route exact path={match.url} render={() => <ReferralDetails id={match.params.id} {...this.props} />} />
      </Switch>
    )
  }
}

ReferralReadOnlyPage.propTypes = {
  match: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  referralDetails: state.referralDetails,
  isLoading: state.referralDetails.isLoading,
  errorMessage: state.app.errorMessage
})

const mapDispatchToProps = { loadReferralData }

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ReferralReadOnlyPage))
