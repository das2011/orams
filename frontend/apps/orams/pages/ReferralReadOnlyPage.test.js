import React from 'react'
import snapshotRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { ReferralReadOnlyPage } from './ReferralReadOnlyPage'

describe('ReferralReadOnlyPage component', () => {
  it('renders correctly when error message is present', () => {
    const props = {
      loadReferralData: jest.fn(),
      match: { params: { id: 1 } }
    }
    const tree = snapshotRenderer
      .create(
        <MemoryRouter>
          <ReferralReadOnlyPage {...props} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
