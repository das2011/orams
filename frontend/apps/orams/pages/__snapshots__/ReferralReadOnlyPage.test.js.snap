// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferralReadOnlyPage component renders correctly when error message is present 1`] = `
<main>
  <div
    className="row headerSection"
  >
    <div
      className="col-xs-12 col-sm-12"
    >
      <h1
        className="au-display-xl"
      >
        Referral Details
      </h1>
    </div>
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Supplier
      </div>
    </div>
    <div />
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Price
      </div>
    </div>
    <div>
      $undefined incl GST
    </div>
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Date of creation
      </div>
    </div>
    <div>
      Invalid Date
    </div>
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Location
      </div>
    </div>
    <div>
      undefined undefined
    </div>
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Created By
      </div>
    </div>
    <div />
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Referral ID
      </div>
    </div>
    <div />
  </div>
  <div
    className="row detailsRow"
  >
    <div
      className="col-sm-3 col-xs-12"
    >
      <div
        className="title"
      >
        Agency
      </div>
    </div>
    <div />
  </div>
  <div
    className="horizontalSeparatorLine"
  />
  <div
    className="row"
  >
    Disclaimer:
  </div>
</main>
`;
